hazelcast/src/main/java/com/hazelcast/jet/impl/JetServiceBackend.java

```diff
@@ -379,4 +406,58 @@ public void startScanningForJobs() {
         jobCoordinationService.startScanningForJobs();
     }
 
+    // Store the metadata about the job jar that is uploaded from client side
+    public void storeJobMetaData(JobMetaDataParameterObject parameterObject) {
+        checkIfCanExecuteJar();
+        jobUploadStore.processJobMetaData(parameterObject);
+    }
+
+    // Store a part of job jar that is uploaded
+    public JobMetaDataParameterObject storeJobMultiPart(JobMultiPartParameterObject parameterObject) {
+        JobMetaDataParameterObject result = null;
+        try {
+            result = jobUploadStore.processJobMultipart(parameterObject);
+        } catch (Exception exception) {
+            jobUploadStore.remove(parameterObject.getSessionId());
+            sneakyThrow(exception);
+        }
+        return result;
+    }
+
+    private void checkIfCanExecuteJar() {
+        if (!jetConfig.isResourceUploadEnabled()) {
+            throw new JetException("Resource upload is not enabled");
+        }
+    }
+
+    // Run the given jar as Jet job. Triggered by both client and member side job uploads
+    public void executeJar(JobMetaDataParameterObject parameterObject) {
+
+        logger.info("executeJar is called for session id :" + parameterObject.getSessionId());
+
+        checkIfCanExecuteJar();
+
+        try {
+            HazelcastBootstrap.executeJar(this::getHazelcastInstance,
+                    parameterObject.getJarPath().toString(),
+                    parameterObject.getSnapshotName(),
+                    parameterObject.getJobName(),
+                    parameterObject.getMainClass(),
+                    parameterObject.getJobParameters(),
+                    true
+            );
+        } catch (Exception exception) {
+            logger.severe("executeJar caught exception when running the jar", exception);
+            // We can not delete the job jar because it is already loaded. If we try to delete it
```

Comment:
Maybe mention that files are cleared by the store

Reply:
The jobs are not cleared. The uploaded jar files will stay so that as the next enhancement the job can be submitted without upload